seafile: flaky renew token test

Occasionally we are seeing a test failure in the seafile backend `renew_test.go`.

https://github.com/rclone/rclone/actions/runs/4305684219/jobs/7508475384

```
panic: sync: negative WaitGroup counter

goroutine 38 [running]:
sync.(*WaitGroup).Add(0xc000332d70?, 0x0?)
	/Users/runner/hostedtoolcache/go/1.20.1/x64/src/sync/waitgroup.go:62 +0xe5
sync.(*WaitGroup).Done(...)
	/Users/runner/hostedtoolcache/go/1.20.1/x64/src/sync/waitgroup.go:87
github.com/rclone/rclone/backend/seafile.TestRenewal.func1()
	/Users/runner/work/rclone/rclone/backend/seafile/renew_test.go:27 +0x33
github.com/rclone/rclone/backend/seafile.(*Renew).renewOnExpiry(0xc00031a440)
	/Users/runner/work/rclone/rclone/backend/seafile/renew.go:36 +0x7e
created by github.com/rclone/rclone/backend/seafile.NewRenew
	/Users/runner/work/rclone/rclone/backend/seafile/renew.go:28 +0x116
FAIL	github.com/rclone/rclone/backend/seafile	0.245s
```

Have you got time to look at this @creativeprojects as I think it is in the code you did recently.

Thank you :-)

--- a/backend/seafile/renew_test.go
+++ b/backend/seafile/renew_test.go
@@ -1,7 +1,6 @@
 package seafile
 
 import (
-	"sync"
 	"sync/atomic"
 	"testing"
 	"time"
@@ -17,19 +16,19 @@ func TestShouldAllowShutdownTwice(t *testing.T) {
 	renew.Shutdown()
 }
 
-func TestRenewal(t *testing.T) {
+func TestRenewalInTimeLimit(t *testing.T) {
 	var count int64
 
-	wg := sync.WaitGroup{}
-	wg.Add(2) // run the renewal twice
-	renew := NewRenew(time.Millisecond, func() error {
+	renew := NewRenew(100*time.Millisecond, func() error {
 		atomic.AddInt64(&count, 1)
-		wg.Done()
 		return nil
 	})
-	wg.Wait()
+	time.Sleep(time.Second)
 	renew.Shutdown()
 
-	// it is technically possible that a third renewal gets triggered between Wait() and Shutdown()
-	assert.GreaterOrEqual(t, atomic.LoadInt64(&count), int64(2))
+	// there's no guarantee the CI agent can handle a simple goroutine
+	renewCount := atomic.LoadInt64(&count)
+	t.Logf("renew count = %d", renewCount)
+	assert.Greater(t, renewCount, int64(0))
+	assert.Less(t, renewCount, int64(11))
 }

